app/src/main/java/org/metabrainz/mobile/viewmodel/SearchViewModel.java

```diff
@@ -0,0 +1,132 @@
+package org.metabrainz.mobile.viewmodel;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import org.metabrainz.mobile.api.data.search.entity.Artist;
+import org.metabrainz.mobile.api.data.search.entity.Event;
+import org.metabrainz.mobile.api.data.search.entity.Instrument;
+import org.metabrainz.mobile.api.data.search.entity.Label;
+import org.metabrainz.mobile.api.data.search.entity.Recording;
+import org.metabrainz.mobile.api.data.search.entity.Release;
+import org.metabrainz.mobile.api.data.search.entity.ReleaseGroup;
+import org.metabrainz.mobile.repository.SearchRepository;
+
+import java.util.List;
+
+public class SearchViewModel extends ViewModel {
```

Comment:
ViewModel to store persistent data and query repository to do search.

Reply:
Marvelous! Good decision of implementing latest Android features

Reply:
Thanks.